libpromises/bootstrap.c

```diff
@@ -116,20 +118,39 @@ bool WriteAmPolicyHubFile(bool am_policy_hub)
  * policy_server.dat file has been removed. Then this function will be called
  * with NULL as new_policy_server, and cf-serverd will keep running even
  * without a policy server set. */
-void SetPolicyServer(EvalContext *ctx, const char *new_policy_server)
+void SetPolicyServer(EvalContext *ctx, const char *host, const char* port)
 {
-    if (new_policy_server)
+    if (host != NULL)
     {
-        xsnprintf(POLICY_SERVER, CF_MAX_IP_LEN, "%s", new_policy_server);
-        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "policy_hub", new_policy_server, CF_DATA_TYPE_STRING, "source=bootstrap");
+        xsnprintf(POLICY_SERVER, CF_MAX_IP_LEN, "%s", host);
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "policy_hub",      host, CF_DATA_TYPE_STRING, "source=bootstrap");
+        if (port != NULL && port[0] != '\n')
+        {
+            EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "policy_hub_port", port, CF_DATA_TYPE_STRING, "source=bootstrap");
```

Comment:
Try to keep new code no more than 78 cols wide. We have tons of old code of course, no need to change what was written years ago.